### DISCUSSION TITLE
Github actions: Stuck in "Expected — Waiting for status to be reported"

### DISCUSSION THREAD

**Discussion by AlexMachin1997** [community/community#26698] [Actions]
<p>Hi everyone,</p>
<p>So I’ve just raised a Pull Request for a personal project, I left it about a day and came back but in my Pull Request it still shows for my unit test related unit test’s. For my unit test workflow actions it’s stuck showing “Expected — Waiting for status to be reported” as seen in the screenshot below.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/0/d/0d832df3deb6bf139b2a3ddf0fc53c7d0e2eb4f3.png" data-download-href="/uploads/short-url/1VxfZnDEILlDcXUvBds5weaQAf1.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/29917195/181099281-574697d5-f203-40e0-a0af-391268e6ade3.png" alt="image" data-base62-sha1="1VxfZnDEILlDcXUvBds5weaQAf1" width="690" height="388" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/0/d/0d832df3deb6bf139b2a3ddf0fc53c7d0e2eb4f3_2_690x388.png, https://aws1.discourse-cdn.com/github/optimized/3X/0/d/0d832df3deb6bf139b2a3ddf0fc53c7d0e2eb4f3_2_1035x582.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/0/d/0d832df3deb6bf139b2a3ddf0fc53c7d0e2eb4f3_2_1380x776.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/0/d/0d832df3deb6bf139b2a3ddf0fc53c7d0e2eb4f3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1920×1080 252 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>Based on what I can see there are certain test’s aren’t running, no logs or no errors it just hasn’t run the remaining unit test’s.</p>
<p>I’m not entirely sure what’s going wrong, it would be great if someone could assist me with my issue.</p>
<p>Thank’s,</p>
<p>Alex</p>

**Comment by airtower-luna** [comment]
<p>Have those workflows been triggered for that PR? If a check is required but that check doesn’t get triggered (e.g. because of a path filter, or because it has been renamed) the “expected check” will block forever.</p>

**Reply by 16oeahr** [reply]
For a workflow with path filter, how to make the workflow as required if it is triggered? i.e. Once it is triggered, how to block PR merge if it fails?

**Reply by darvid7** [reply]
+1 same question!

**Reply by p-flock** [reply]
https://github.com/orgs/community/discussions/26698#discussioncomment-6816209

having the same issue

**Reply by danielo515** [reply]
That is definitely a bug. Why would you want a status check that will never run to be required? Is there a way to ensure only make it required when the rules of that check will make it run?

**Reply by marcellodesales** [reply]
Also, even adding branch rules to the branch from the PR we can't dismiss or bypass the branch settings... First bug in Github in many years... I've tried everything and I need a github organization workflow to be run in all PRs... I can't see where the runner is being executed, since I own the Github Org and I own the Gitub action Runner instances in the org... Here's what I tried

- Defined myself as users who can dismiss PRs
- Defined myself as an Admin/Owner of the repository
- Re-created the PR with another branch

**On top of that, there's no links to the actual action run so that we can verify what's been waiting for...!**

<img width="963" alt="Screenshot 2023-11-04 at 1 36 20 PM" src="https://github.com/community/community/assets/131457/ee3b77a4-f400-4f92-a360-a94650ef3cee">

**Comment by AlexMachin1997** [comment]
<p>Yeah the action’s are triggered whenever a Pull Request is created.</p>
<p>There is no special filters within the job from what I can see in the workflow files, the only filters is the <code>on: [pull_request]</code></p>
<p>Just in case I’ve done something unintentional <a href="https://github.com/AlexMachin1997/TheOpenMovieDB-React-Example/blob/11291a5f208cab6768a3f242d894c6a10154c927/.github/workflows/test-action.yml" rel="noopener nofollow ugc">here’s</a> the link to the workflow action.</p>
<p>What’s a little strange is that when I go to the “Action’s” tab I can see all the workflow actions did in fact run (see screenshot attached below) however in the Pull Request I raised the status is still stuck in that “Expected - Waiting for status to be reported” state.</p>
<p>Screenshot of the “Actions” tab. FYI The Pull Request is called Styled-Components to Tailwind CSS<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/e/9/e9dce8a1724b3ef5d37e27f5eaf5c300281cd436.png" data-download-href="/uploads/short-url/xmQxOkhnIKvRXQOpLA3EXfjUAhU.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/29917195/181099282-b0b93bd2-8474-4cf4-8def-3fa9bf491689.png" alt="image" data-base62-sha1="xmQxOkhnIKvRXQOpLA3EXfjUAhU" width="690" height="388" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/e/9/e9dce8a1724b3ef5d37e27f5eaf5c300281cd436_2_690x388.png, https://aws1.discourse-cdn.com/github/optimized/3X/e/9/e9dce8a1724b3ef5d37e27f5eaf5c300281cd436_2_1035x582.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/e/9/e9dce8a1724b3ef5d37e27f5eaf5c300281cd436_2_1380x776.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/e/9/e9dce8a1724b3ef5d37e27f5eaf5c300281cd436_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1920×1080 275 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by airtower-luna** [comment]
<p>I don’t see any workflow runs by those exact names in the Actions tab of that repository, it and it looks like the “Application testing / Running React-Testing-Library and Jest tests” job has been changed to use Node 16 only now, instead of 12.x and 14.x.</p>
<p>I suspect you might need to update the list of required checks (jobs) in the repository configuration, at least if those changes to the jobs were intentional.</p>

**Reply by juampynr** [reply]
That was my case! I did change the job names. Thanks for the suggesion.

**Comment by AlexMachin1997** [comment]
<p>Hmm that is strange.</p>
<p>The last time I updated my project I swapped out the “matrix” functionality which can on version 12 and 14 though since then it’s been set to run on only v16 as seen in the screenshot</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/c/f/cf93593ab8b38063f35cbaff7c7bf2382a469597.png" data-download-href="/uploads/short-url/tCiwOfepC0G0q1tiLAEKEtxP9Ub.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/29917195/181099283-16b77981-c878-4998-851f-c9faa05c789d.png" alt="image" data-base62-sha1="tCiwOfepC0G0q1tiLAEKEtxP9Ub" width="690" height="388" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/f/cf93593ab8b38063f35cbaff7c7bf2382a469597_2_690x388.png, https://aws1.discourse-cdn.com/github/optimized/3X/c/f/cf93593ab8b38063f35cbaff7c7bf2382a469597_2_1035x582.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/c/f/cf93593ab8b38063f35cbaff7c7bf2382a469597_2_1380x776.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/f/cf93593ab8b38063f35cbaff7c7bf2382a469597_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1920×1080 356 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>.
<p>I’ve also updated the workflow action versions numbers as you suggested though the workflow actions are still behaving the same as before sadly.</p>

**Comment by AlexMachin1997** [comment]
<p>Hmm seem’s like it’s wait for actions which simply don’t exist and haven’t for a while.</p>
<p>I can’t seem to find them in the “Actions” tab but it’s not appearing.</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="AlexMachin1997" data-post="6" data-topic="247329">
<div class="title">
<div class="quote-controls"></div>
 AlexMachin1997:</div>
<blockquote>
<p>Hmm seem’s like it’s wait for actions which simply don’t exist and haven’t for a while.</p>
</blockquote>
</aside>
<p>That’s exactly my point. Those jobs still seem to be configured as required in your branch protection rules, that’s what the “Required” label means. When you rename or remove jobs, you need to update any branch protection rules that use them.</p>

**Comment by AlexMachin1997** [comment]
<p>I’m confused now.</p>
<p>The changes where I specify the “NodeJS” version were merged months ago.</p>
<p>I’ve just created a new Pull Request with the updated github actions and I’m experiencing the same issue with the " Styled-Components to Tailwind CSS" :confused:</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>I’m talking about <a href="https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches">branch protection rules</a>. I can’t see exactly which rules exist in your repository (that’s part of the repository settings, so not available to me as a random other user), but I can see those checks are shown as “required” in the PR. That means there is a branch protection rule that says those jobs must have passed before the PR can be merged.</p>
<p><a href="https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule">Configuration of branch protection rules</a> is completely separate from your workflows. I assume that at some point in the past you’ve marked the previous jobs as required checks (step seven in the documentation, or search for “required status checks”). Once enabled, those jobs are required <em>by name</em>.</p>
<p>You’ve changed job names, so you need to update the branch protection rule: Remove the old jobs, so waiting for jobs that don’t exist any more doesn’t block merging. And ideally add your new job (the non-matrix one) to that gets checked.</p>

**Reply by c5k4** [reply]
Not really answer the 'Expected' question. Prisma Cloud / Code Aanlysis check normally runs within 15-20 minutes for the repo. Occasionally will get stuck in the Expected state. What might be going on and more importantly, how to get out of it to Merge the PR, without having to override it, using the merge without waiting check box.

**Reply by piotrekkr** [reply]
I don't think there is a way to force rerun just one **stuck** check. This requires PR event to be fired again and you can't artificially trigger events in GitHub. You will need to trigger them by interacting with GitHub Actions either by UI or API. If some checks are stuck and are triggered by `pull_request` events then you can push empty commit to pr branch. This should rerun them even if some are stuck. 

Also some external apps may use checks API to add checks results manually when you trigger some events in GH. If they are stuck or fail to communicate with GH API, then maybe they offer some rerun function on their side.

**Reply by bayer-veera** [reply]
Will empty commits consider as new commits and require to get approval again?

**Reply by piotrekkr** [reply]
Yeah. Adding empty commit triggers PR checks

**Comment by AlexMachin1997** [comment]
<p>Ah gotcha.</p>
<p>So even though I updated the workflow actions I needed to modify the actual branch rules which were still present.</p>
<p>Annoying but less learned.</p>
<p>Thank’s for sticking with me <a class="mention" href="/airtower-luna">@airtower-luna</a></p>

**Comment by divyachowd** [comment]


**Reply by divyachowd** [reply]
![image](https://user-images.githubusercontent.com/118965543/203727553-0de7ce33-5bde-480d-90c9-7ae7f8f9b6e0.png)

**Comment by koo5** [comment]
In my case, there was a `:` somehow accidentally tacked onto the required job name. Not easy to figure out!

**Comment by p-flock** [comment]
following on from [this question](https://github.com/orgs/community/discussions/26698#discussioncomment-5944210):
I have a required test akin to `check_file_tests_if_file_changed` which then runs a workflow file with multiple test jobs, 
```yaml
name: Run all tests for file

jobs:
  test1:

  test2:

  test3

```

when I set a branch protection rule to prevent merging until `check_file_tests_if_file_changed` has run, the check hangs and never gets a status reported even if the 3 child jobs pass.

But if I require the 3 child jobs individually, they will prevent merging to the protected branch even if they were skipped because there were no file changes

**Comment by koppor** [comment]
In my case, I had "path filters" on the PR. I wanted the deployment only happen if only tests were changed (`src/main/test/**` was ignored when triggering). However, we marked working deployment as a must. Thus, even if not triggered in some cases, it was required.

Removing any filter in the trigger solved the issue here.

For reference: Required changes at https://github.com/JabRef/jabref/pull/10319; other information linked there.

**Comment by Diapolo10** [comment]
I've run into a similar issue. A community contributor put forth a pull request (https://github.com/Diapolo10/iplib3/pull/105) to a project I'm maintaining, but the unit tests and linter do not run in the actual pull request.

They work fine for the individual commits, but I'd rather not force the pull request to go through manually.

**Reply by koppor** [reply]
I checked. The trigger `pull_request` was missing. Without that trigger, there is no check, even if the contributor "pushed". The difference is that "push" is in the repository itself - and (AFAIK) ignores activity of contributors in a pull request.

I submitted a PR fixing it at https://github.com/Diapolo10/iplib3/pull/107.

**Reply by tomsit-ionos** [reply]
not sure if i understood here correctly, but an `on.pull_request` is not needed per see. The only thing needed AFAIKS is that the required check-job is executed somehow as part of a push the the head of the PR, i.e. for me an `on.push` is sufficient

**Comment by dyedwiper** [comment]
In my case there were **merge conflicts** and it turned out that "Workflows will not run on pull_request activity if the pull request has a merge conflict. The merge conflict must be resolved first.", like stated here: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

**Comment by thisisthemurph** [comment]
In my case, this seemed to be a transient issue where the workflow didn't run (couldn't see it in the list of checks) but the result was still being awaited. I ran the following commands to force another push to the PR:

```
git commit --amend --no-edit
git push --force-with-lease
```

**Reply by BijieXu** [reply]
Nice, works for me.

**Comment by raulelba** [comment]
I'm seeing this issue quite often. In my case it seems to be a happening when we amend a commit and force push to the branch while there are previous jobs still running. The action runs are cancelled, but sometimes they stay lingering their "waiting for status to be reported" and we cannot do anything to re-trigger them. I have tried everything that crossed my mind, from re-running the actions to force-pushing, but the status of the check won't change. When this happens in a check that's required it prevents the PRs from being merged.
 
![1](https://github.com/community/community/assets/141129330/f9bfe0e6-4b50-4afc-942d-5bfdac168470)
![2](https://github.com/community/community/assets/141129330/9c89500e-b5ba-4dcd-acb2-8ab21f6d7df5)
![3](https://github.com/community/community/assets/141129330/a4545455-aa57-4357-9bf8-44212000755d)

**Comment by BlaineEXE** [comment]
I am encountering this issue here also https://github.com/rook/rook/pull/14299, but in this case, there was no force-push that preceded it. It seems to have gotten stuck all on its own. I tried a force-push to get the test to rerun, to no avail.

It would be really nice to at least be able to re-run this check when it gets into stuck state like this. As it is, it seems like it's handled via some special mechanism that doesn't show up in github actions, and that's why it's blocking PRs randomly.

**Reply by BlaineEXE** [reply]
We have started to see this behavior regularly over the past week (more than 5 times) in github.com/rook/rook. Is GitHub aware of any ongoing issues with the DCO check?

**Comment by Galkon** [comment]
Feels like we are running into this every other PR / push we make -- incredibly frustrating. Is GitHub going to fix this?

**Comment by stevenh** [comment]
Seeing this on PR's which have worked flawlessly, seems to be no rhyme or reason. Example https://github.com/testcontainers/testcontainers-go/pull/2664

**Comment by jarvis8x7b** [comment]
is there currently a fix for this? we're getting it on every PR now, ideally want to have it work properly rather than having to bypass branch protection rules
<img width="983" alt="image" src="https://github.com/user-attachments/assets/3ba643d7-e4bd-479c-8854-f733c059b016">

**Reply by marcellodesales** [reply]
Same behavior... the rule has everything correctly set and it hangs in `Expected - waiting for workflow to run`

**Comment by stevenh** [comment]
When we see this, its all actions but not all PR's, very rarely triggers.

Given this I would recommend checking that your branch protection rules don't require an old or renamed action @AlexMachin1997 as this is what you'll see in that case.

**Comment by WowSuchRicky** [comment]
For anyone who arrives to this more recently, it was indeed also an old job with the same name that was expected as running but did not ever start due to the fact that it doesn't exist anymore. Try removing it from the branch protection as others have suggested, and then re-add it (after a refresh or similar, to make sure the old job ID isn't cached). This solved the issue completely for me. 

Also worth mentioning that the branch protection actions list seems to populate from main/dev only, at least in our setup, so I would make sure you have updated those appropriately.

**Reply by alexlovar** [reply]
It would be great if the detailed instructions were included too

**Comment by nickguo** [comment]
For anyone arriving here in the future, my Stuck action problem was resolved by closing and then immediately re-opening the PR.

**Reply by handipriyono** [reply]
Thanks, it works for me too.!  
Close and then re-Open the Pull Request really works and trigger re-running

**Reply by arielshaqed** [reply]
Is there a better workaround?  My merge to master runs a 15-minute CI for [this particular PR](https://github.com/treeverse/lakeFS/pull/8268) (hint: testing on Azure is slow), and I prefer to wait less in future.

**Reply by nickguo** [reply]
@arielshaqed No better workaround to my understanding because if the PR is stuck on "Waiting for status...", most likely your CI never triggered in the first place.

**Reply by deeno35** [reply]
In case anyone is wondering, closing/re-opening the PR works because it causes all of the tests to re-run.  Even ones that had passed already.

If any GH devs stop by, it should be possible to cancel a job that is in this state to avoid having to re-run them all as a workaround.

**Comment by lemon1964** [comment]
test Expected — Waiting for status to be reported

The rule is on the main branch.
And the process is on the develop branch.

Changed main to develop in the rules.
The rule check passes immediately

**Comment by ahmedrafayat** [comment]
In my case, my feature branch was stale. So I merged it with the latest `main` and its working again.

**Comment by coreybyrum7** [comment]
My repository contained a ruleset that required the workflow to pass before merging. However, the said workflow contained a filter via the `on:` key ... the key allowed me only run the workflow under certain conditions.

The _hanging/missing GHA_ was only because the workflow as never triggered, as the PR contained commits that didn't fall into the filter.

Hope this helps

**Comment by lzm0** [comment]
In my case, I resolved it by rebasing on top of the main branch.

**Comment by seangwright** [comment]
This could also happen because you have specific file patterns in your triggers for a workflow/jobs and you've created a PR with changes that don't match those patterns.

Example: With this workflow combined with a branch that requires "Build App" completes and PR that only changes an `.md` file.

```yml
name: "CI: Build and Test"

on:
  pull_request:
    branches: [main]
    paths:
      - "**.js"
      - "**.json"
jobs:
  build-app:
  name: Build App
  
  ...
```

The workflow won't execute but the PR will continue to wait for it to complete.

**Reply by heyitsaamir** [reply]
This was my issue. Not totally sure how to get past this?

**Reply by piotrekkr** [reply]
You can do a workaround with running check always but inside check steps you would use an action to check what changed. If files you want changed then run the rest of the steps. If not, you can skip remaining steps. Example action you could use is https://github.com/tj-actions/changed-files . You can also use git diff but this is more tricky to use with Github.

**Comment by noodlecollie** [comment]
For me, my particular issue was that "Job name" did not match when using a matrix strategy, as the matrix strategy appends the parameters in brackets to each job name. This means I actually had to look for status checks named "Job name (param1)" and "Job name (param2)", etc.

The reliable rule I was able to follow was that for a pull request, where a job is displayed with the format `Workflow file / Job name (matrixparam1, matrixparam2) (pull_request)`, the actual name that needed to be provided as a required status check was the text between `/` and `(pull request)` (non-inclusive).

<img width="811" height="235" alt="image" src="https://github.com/user-attachments/assets/2830f6b5-4d6e-47ba-9386-f20be23ced91" />
